services/tracing.py

```diff
@@ -136,6 +136,10 @@ def wrap_with_span(*args, **kwargs):
                         )
                     attributes_dict[attribute] = str(func_arg)
 
+                # Add attributes to the current (parent) span before we start the child span
+                span = trace.get_current_span()
+                span.set_attributes(attributes_dict)
+
```

Comment:
Worth testing the attributes are set on the both the parent and childspan in test_tracing?